core/commands/commit.py

```diff
@@ -95,15 +99,38 @@ def run_async(self):
         if self.view.settings().get("git_savvy.commit_view.include_unstaged"):
             self.add_all_tracked_files()
 
+        commit_args = ["commit", "-q"]
+
         if self.view.settings().get("git_savvy.commit_view.amend"):
-            self.git("commit", "-q", "--amend", "-F", "-", stdin=commit_message)
-        else:
-            self.git("commit", "-q", "-F", "-", stdin=commit_message)
+            commit_args.append("--amend")
+
+        if self.signed_commit:
+            commit_args.append("--gpg-sign")
+
+        commit_args.extend(["-F", "-"])
+
+        self.git(*commit_args, stdin=commit_message)
```

Comment:
I've used the following pattern elsewhere (the `git` method automatically strips falsey args).  Please use it here.

```python
self.git(
    "commit",
    "-q",
    "--amend" if self.view.settings().get("git_savvy.commit_view.amend") else None,
    "--gpg-sign" if self.signed_commit else None,
    "-F"
    "-"
    stdin=commit_message
    )
```